### PR TITLE
Fix disabled blocks logical error on Widgets screen

### PR DESCRIPTION
**Pull request by getdave**
## Description
[As @draganescu has noticed](https://github.com/WordPress/gutenberg/pull/32761#discussion_r703704489) the logic which disables blocks in Widgets will throw an error if `ALLOW_REUSABLE_BLOCKS` is `true`. This is because the result of the expression was not always iterable.

This PR fixes that by ensuring the result of the expression is always iterable. 

## How has this been tested?

1. Open Widgets screen. Check there are no console errors and that you cannot use Resuable blocks.
2. Go to `packages/edit-widgets/src/constants.js` and change `ALLOW_REUSABLE_BLOCKS` to `true`.
3. Rebuild the Plugin.
4. Go to Widgets screen. Check there are no console errors and that you can use Resuable blocks.

## Types of changes
Bug fix (non-breaking change which fixes an issue) 

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/edit-widgets/src/index.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ import {
 const disabledBlocks = [
 	'core/more',
 	'core/freeform',
-	...( ! ALLOW_REUSABLE_BLOCKS && [ 'core/block' ] ),
+	...( ! ALLOW_REUSABLE_BLOCKS ? [ 'core/block' ] : [] ),
```
Comment (kevin940726) [inline packages/edit-widgets/src/index.js]: Nit: Would be slightly easier to read if we negate the ternary.

```suggestion
	...( ALLOW_REUSABLE_BLOCKS ? [] : [ 'core/block' ] ),
```

**Review by kevin940726 — APPROVED** [review]
Thanks!

File: packages/edit-widgets/src/index.js (hunk 1 above)
Comment (getdave) [reply packages/edit-widgets/src/index.js]: Probably. I have a weird thing where I don't like having the "empty" bit first but 🤷 

Perhaps changing the _constant_ to `DISABLE_RESUABLE_BLOCKS` would make this whole thing make more sense?

Comment (kevin940726) [reply packages/edit-widgets/src/index.js]: I don't mind either way 😆.

Comment (Mamaduka) [reply packages/edit-widgets/src/index.js]: >  I have a weird thing where I don't like having the "empty" bit first but 🤷

I do that all the time as well 😆

**Review by adamziel — APPROVED** [review]

**Pull request merged**